### PR TITLE
Feat: retry fulfillCondition automatically

### PR DESCRIPTION
**Pull request by sharafian**
If `fulfillCondition` fails in the `fulfill` callback, it will keep retrying until the transfer's expiry. If it still has not succeeded at that point, it will reject.

### CODE REVIEW

File: src/lib/transport.js [hunk 1]
```diff
@@ -137,7 +139,12 @@ function * listen (plugin, {
         destinationAccount,
         destinationAmount,
         fulfill: function () {
-          return plugin.fulfillCondition(transfer.id, fulfillment)
+          return retryPromise({
+            callback: () => plugin.fulfillCondition(transfer.id, fulfillment),
+            minWait: minFulfillRetryWait || 250,
```
Comment (emschwartz) [inline src/lib/transport.js]: Defaults should be declared as constants at the top of the file

File: src/lib/psk.js [hunk 2]
```diff
@@ -21,7 +21,9 @@ const assert = require('assert')
   * @param {Buffer} [params.data=null] Additional data to include in the request
   * @param {Object} [params.headers=null] Additional headers for private PSK details. The key-value pairs represent header names and values.
   * @param {Object} [params.publicHeaders=null] Additional headers for public PSK details. The key-value pairs represent header names and values.
-  * @param {Object} [params.disableEncryption=false] Turns off encryption of private memos and data
+  * @param {Boolean} [params.disableEncryption=false] Turns off encryption of private memos and data
+  * @param {Number} [params.minFulfillRetryWait] Minimum amount of time to wait before retrying fulfillment
```
Comment (emschwartz) [inline src/lib/psk.js]: This should say what the default is

File: README.md [hunk 3]
```diff
@@ -302,9 +302,7 @@ SPSP query response
 | minimum_destination_amount | <code>string</code> | Integer string representing the minimum that the receiver will be willing to accept. |
 | ledger_info | <code>Object</code> | An object containing the receiver's ledger metadata. |
 | ledger_info.currency_code | <code>string</code> | The currency code of the receiver's ledger. |
-| ledger_info.currency_symbol | <code>string</code> | The currency symbol of the receiver's ledger. |
-| ledger_info.precision | <code>string</code> | The precision of the receiver's ledger. |
-| ledger_info.scale | <code>string</code> | The scale of the receiver's ledger. |
```
Comment (emschwartz) [inline README.md]: Why is this commit also modifying the precision and scale?

File: test/utilSpec.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 'use strict'
 
 const chai = require('chai')
+const expect = chai.expect
```
Comment (emschwartz) [inline test/utilSpec.js]: Better to use either expect or assert but not both at the same time

File: README.md (hunk 3 above)
Comment (sharafian) [inline README.md]: The commit that modified it just didn't build docs

File: src/lib/transport.js [hunk 5]
```diff
@@ -137,7 +142,12 @@ function * listen (plugin, {
         destinationAccount,
         destinationAmount,
         fulfill: function () {
-          return plugin.fulfillCondition(transfer.id, fulfillment)
+          return retryPromise({
+            callback: () => plugin.fulfillCondition(transfer.id, fulfillment),
```
Comment (emschwartz) [inline src/lib/transport.js]: I still think we _might_ want to catch the final errors defined in the Ledger Plugin Interface so that we don't retry those

**Pull request merged**